gensim/test/test_fasttext.py

```diff
@@ -1300,6 +1306,253 @@ def _run(self, fin):
+    return model
+
+
+def calc_max_diff(v1, v2):
+    return np.max(np.abs(v1 - v2))
+
+
+class SaveFacebookFormatModelTest(unittest.TestCase):
+    """
+    This class containts tests that check the following scenario:
+
+    + create fastText model1
+    + save a model1 to Facebook binary file
+    + load the file as model2
+    + check if  model1 == model2
+    """
+
+    def _check_roundtrip_model_model(self, model_params):
+
+        with temporary_file("roundtrip_model_to_model.bin") as fpath:
+
+            model_trained = _create_and_save_test_model(fpath, model_params)
+
+            model_loaded = gensim.models.fasttext.load_facebook_model(fpath)
+
+            self.assertEqual(model_trained.vector_size, model_loaded.vector_size)
+            self.assertEqual(model_trained.window, model_loaded.window)
+            self.assertEqual(model_trained.epochs, model_loaded.epochs)
+            self.assertEqual(model_trained.negative, model_loaded.negative)
+            self.assertEqual(model_trained.hs, model_loaded.hs)
+            self.assertEqual(model_trained.sg, model_loaded.sg)
+            self.assertEqual(model_trained.trainables.bucket, model_loaded.trainables.bucket)
+            self.assertEqual(model_trained.wv.min_n, model_loaded.wv.min_n)
+            self.assertEqual(model_trained.wv.max_n, model_loaded.wv.max_n)
+            self.assertEqual(model_trained.vocabulary.sample, model_loaded.vocabulary.sample)
+            self.assertEqual(set(model_trained.wv.index2word), set(model_loaded.wv.index2word))
+
+            for w in model_trained.wv.index2word:
+                v_orig = model_trained.wv[w]
+                v_loaded = model_loaded.wv[w]
+
+                try:
+                    self.assertLess(calc_max_diff(v_orig, v_loaded), MAX_WORDVEC_COMPONENT_DIFFERENCE)
+                except AssertionError as e:
+                    s = e.args[0]
+                    s += " (max difference for components in wordvector of \"%s\"" % w
+                    s += " larger than the thershold %f" % MAX_WORDVEC_COMPONENT_DIFFERENCE
+                    e.args = (s,)
+                    raise e
+
+    def test_round_trip_model_model_skipgram(self):
+        model_params = {
+            "size": 10,
+            "min_count": 1,
+            "hs": 1,
+            "sg": 1,
+            "negative": 5,
+            "seed": 42,
+            "workers": 1}
+        self._check_roundtrip_model_model(model_params)
+
+    def test_round_trip_model_model_cbow(self):
+        model_params = {
+            "size": 10,
+            "min_count": 1,
+            "hs": 1,
+            "sg": 0,
+            "negative": 5,
+            "seed": 42,
+            "workers": 1}
+        self._check_roundtrip_model_model(model_params)
+
+
+def _read_binary_file(fname):
+    with open(fname, "rb") as f:
+        data = f.read()
+    return data
+
+
+class SaveFacebookFormatFileGensimTest(unittest.TestCase):
```

Comment:
There are really four things you need to know about this class:

0. It's a test
1. It's for the save function
2. It compares results byte to byte
4. It tests pure gensim (unlike it's FB cousin)

```suggestion
class SaveGensimByteIdentityTest(unittest.TestCase):
```

Everything else, e.g.

- FacebookFormat
- File

isn't really relevant enough to warrant putting it in the very name.

Reply:
Corrected.